src/main.rs

```diff
@@ -219,6 +220,19 @@ impl LanguageServer for Backend {
 
         Ok(help)
     }
+
+    async fn formatting(&self, params: DocumentFormattingParams) -> Result<Option<Vec<TextEdit>>> {
+        let _ = params;
```

Comment:
There should be a `uri` somewhere in `params` that says which document should be formatted. It's possible now to get that document rather than just `main`.